Pull request for V1 of my Ecommerce Application

Akshaay9 opened the pull request:
Hello Everyone,

This is version one of my e-commerce application Fitt Shark.

Website is deployed  live on - https://fitt-sharkk-v1.netlify.app/

Tech Stach -

Backend - I used Express JS for  backend framework, node for runtime environment, Mongo DB for database, and jwt for authentication.
Frontend - I used React.JS for JS library, CSS for styling, context API for state management, and react-router v-6 for routing.

Features :

 1. Responsive Landing page with 6 different  featured products.
 2. Features including add to cart, add to wishlist and update cart and wishlist  items integrated with real-time database.    
 3.  5 Different types of product filter option to filter products according to users need
 4.  individual page for each product to understand complete product details
 5.  Search option to search from 70+ products
 6.  Login and sign up with real-time authentication fully integrated with backend, database and jwt for tokens and 
    authentication.
 7.  Final checkout page with all products lists, selected address and payment option
 8.  Address management page for adding/updating user address
 9.  Payment gateway has been integrated with stripe payment option
10. Checkout page with the complete list of cart items along with the total price and policies. 
11.  Order success page with all ordered product list, payment,address and delivery information visible after users payment
12. Users private profile/page for updating password and to view previous purchase history and open indiviual purchase history.

Please go through the site and feel free to share any feedback

Thanks & Regards,
Akshay

Akshaay9 committed:
trying filteres for visibility

Akshaay9 committed:
trying filteres for visibility for price filetrs

Akshaay9 committed:
successfully added four different filters with styling and animation using react hoocks

Akshaay9 committed:
added animation to filter bar

Akshaay9 committed:
Refactored Footer Component

Akshaay9 committed:
made changes to postioning of the filter bars so that they cannot push other elements out of their position

Akshaay9 committed:
added womens new release Banner/Poster with description and mobile responsiveness

Akshaay9 committed:
added gym accesories poster with call to action buttons

Akshaay9 committed:
fixed navBar dropdown link positioning

Akshaay9 committed:
made some UI changes on Home screen and reDesigned Product lists

Akshaay9 committed:
added Screen and Poster for Home Workout Equipments

Akshaay9 committed:
added Resistance Training Equipment Page And poster

Akshaay9 committed:
created the list products to be added in the MensNewDrop/MensCLothing page

Akshaay9 committed:
created folder structure for contextAPI, cart and wishlist components

Akshaay9 committed:
feture-Added Usecontext for routing between multiple pages

Akshaay9 committed:
Feature- Added product list of mensNewDrop,Create a useReducer and context Api for the same

Akshaay9 committed:
Feature: Added add to cart,increase, descrease qty and delete from cart

Akshaay9 committed:
added Filter Feature to the mensNewDropPage

Akshaay9 committed:
added working filter

Akshaay9 committed:
Added filter items successfully

Akshaay9 committed:
working on cart component

Akshaay9 committed:
Added  UI to cart Items and Added ADD/Remove buttons on cart

Akshaay9 committed:
added localstorage to persist cart Items

Akshaay9 committed:
added Wish List Functionlity

Akshaay9 committed:
added localStorage to persist state on wishlist

Akshaay9 committed:
adding single product display page

Akshaay9 committed:
UI Bug Fix

Akshaay9 committed:
UI Bug Fix

Akshaay9 committed:
added single product view feature and made it responsive

Akshaay9 committed:
added total price and total quantity of items present in the cart

Akshaay9 committed:
added data and product list to the app

Akshaay9 committed:
added products and filter page for womens cloths

Akshaay9 committed:
added data

Akshaay9 committed:
configured new product list

Akshaay9 committed:
bug fix

Akshaay9 committed:
added some fix to bugs

Akshaay9 committed:
added Home workout equipments

Akshaay9 committed:
added resistanceproducts

Akshaay9 committed:
added routing to whole application

Akshaay9 committed:
added  yoga and gym accessories products

Akshaay9 committed:
added  single product viewer through routing

Akshaay9 committed:
added search bar to application

Akshaay9 committed:
added  cta to wishList

Akshaay9 committed:
refactored API calls and created a utility class to get all request

Akshaay9 committed:
refactored whole productListing code and reduced a code to 50%

Akshaay9 committed:
refracored code of prodict lsit

Akshaay9 committed:
refactored code

Akshaay9 committed:
added responsive navbars for mobile

Akshaay9 committed:
made some UI changes and added some details regarding the Items

Akshaay9 committed:
made filter bars responsive

Akshaay9 committed:
made all filter bars responsive

Akshaay9 committed:
added footer and made it responsive

Akshaay9 committed:
added svg and UI improved

Akshaay9 committed:
navUI fix

Akshaay9 committed:
added toat

Akshaay9 committed:
made some UI changes and BUG fix

Akshaay9 committed:
made some UI changes

Akshaay9 committed:
refractored whole filter component and made utility functions tomanage filtering of data

Akshaay9 committed:
bug fix

Akshaay9 committed:
made SignUp route and added signup page

Akshaay9 committed:
added login and signup page with validation

Akshaay9 committed:
refractored whole filter components and removed all useState and useEffect to make code more faster and efficient

Akshaay9 committed:
made some UI fix

Akshaay9 committed:
Started adding backend

Akshaay9 committed:
made some changes to .json file

Akshaay9 committed:
Server error fix

Akshaay9 committed:
Server error fix:1

Akshaay9 committed:
server bug fix:2

Akshaay9 committed:
revamping whole backend

Akshaay9 committed:
some UI bug fix

Akshaay9 committed:
Started working on backend

Akshaay9 committed:
Backend BUG fix

Akshaay9 committed:
Backend BUG fix:1

Akshaay9 committed:
Backend bug fix:3

Akshaay9 committed:
Backedn bug fix:3

Akshaay9 committed:
Backedn bug fix:4

Akshaay9 committed:
Backedn bug fix:5

Akshaay9 committed:
Added seeder files and imported all the product list and added few USers and admins to the data base

Akshaay9 committed:
added backend route to fetch all the products

Akshaay9 committed:
created login and signup route, added auth and assigned JWT to logged user

Akshaay9 committed:
added Backend Route for Cart products and update/delete cart items

Akshaay9 committed:
configured config file

Akshaay9 committed:
added checkout route

Akshaay9 committed:
bug-fix

Akshaay9 committed:
bug fix config file

Akshaay9 committed:
added .env file

Akshaay9 committed:
Removed config file

Akshaay9 committed:
Removed config file and bug fix

Akshaay9 committed:
bug fix:backend

Akshaay9 committed:
Fixed all backend bugs and added .env files and hosted backend on heroku

Akshaay9 committed:
connected backend to frontend to get all the product list

Akshaay9 committed:
added search through backend and frontend

Akshaay9 committed:
added search through backend

Akshaay9 committed:
ading auhentication to frontend

Akshaay9 committed:
Successfully added front end Login and signup using backend

Akshaay9 committed:
connected attToCart to backend and fixed few bugs

Akshaay9 committed:
connected cart and wishlist to backend and enabled users to increase/decrease cart qty and wishlist products

Akshaay9 committed:
added Shipping pages to front end

Akshaay9 committed:
added adress management in backend and integrated Stripe api for payment processing

Akshaay9 committed:
Added payment gateway successfully and integrated Checkout page on backend

Akshaay9 committed:
added final ordered list page with success indication,payment details and shipping details

Akshaay9 committed:
Added modal for non logged in users to redirect to login

Akshaay9 committed:
added log out option on front and cleared whole cart and wislist once the user logs out

Akshaay9 committed:
JWT error fixing

Akshaay9 committed:
fixed JWT error on backend

Akshaay9 committed:
Backend JWT bug Fix

Akshaay9 committed:
refractored whole API calling funtions for CART and WISHLIST on all Pages

Akshaay9 committed:
added Toasts on All API calls and user actions

Akshaay9 committed:
added loader to all API calls CTA

Akshaay9 committed:
added reset password feature in backend as well as front end

Akshaay9 committed:
added spinner to wishlist and refractored wishlist

Akshaay9 committed:
added spinner to Cart Items

Akshaay9 committed:
added loader to single view of products

Akshaay9 committed:
revamped whole address management system, let people add multiple addresses and select from respectively, added address update,edit and delete both on backend and fornt end successfully

Akshaay9 committed:
added loader to adress management buttons and success payment page plus fe UI bug fix and modification

Akshaay9 committed:
added skeleton loader to all the pages

Akshaay9 committed:
made few UI changes

Akshaay9 committed:
fixed backend bug

Akshaay9 committed:
few few ui bugs

Akshaay9 committed:
fixed login modal height"

Akshaay9 committed:
fixed product filter bug

Akshaay9 committed:
small UI bu fix and made sure app is working fine"

Akshaay9 committed:
added ratings to the review sections and revamped UI of single product view page and added discound on each page

Akshaay9 committed:
made some UI changes on responsiveness of filter bar"

Akshaay9 committed:
added hamberger menu and Fixed some UI

Akshaay9 committed:
SOme UI bug FIx

Akshaay9 committed:
removed CSS of modal and used it from css library

sdhamani commented:
Hi Akshay,

The site is pretty impressive and also the code looks great. I noticed you have added many extra features. The order placement and payment is seamless.

The only change you should do it 
1. I guess you should not show the password suggestion or corrections when the user is trying to log in. You just need to check that while creating an account.
2. You can update the README of this repo.

Also, I have one suggestion for your future projects that your git commits are ambiguous like some bug fixed so many be if you can elaborate them it will look better.

Regards,
Sagar Dhamani

Akshaay9 committed:
Updated Readme.md File

Akshaay9 commented:
> Hi Akshay,
> 
> The site is pretty impressive and also the code looks great. I noticed you have added many extra features. The order placement and payment is seamless.
> 
> The only change you should do it
> 
> 1. I guess you should not show the password suggestion or corrections when the user is trying to log in. You just need to check that while creating an account.
> 2. You can update the README of this repo.
> 
> Also, I have one suggestion for your future projects that your git commits are ambiguous like some bug fixed so many be if you can elaborate them it will look better.
> 
> Regards,
> Sagar Dhamani

Hey Sagar
Thanks you for your feedback
1) I have added password suggestion in login because i just wanted to make users experience better, in case they forget the password type they have used for signing up so they dont have to reset password
2) I have added Readme.md and updated with all the project details
3)Yes, i would try to eloberate some of my git commit with the kind of bug i faced instead of just `bug fix` which helps developer/viewer to understand  my code 

Thank you
Akshay

sdhamani commented:
I guess then you are good to go. Great work.

shivamupadhyay008 commented on Backend/Data/ProductList.js:
```diff
@@ -0,0 +1,2191 @@
+export const ProductList = [
+    {
+        id: 1,
```
if data is coming form db it would be great to removing this file so your code looks clean

Akshaay9 replied:
Hey Shivam
Thank you for your review,
I kept product details locally too, because i was afraid if something happens to my data in database then i would loose all data so,
Appreciate your suggestion
Thank you
Akshay

shivamupadhyay008 commented on src/Contexts/ProductListContext/GymAccessories.js:
```diff
@@ -0,0 +1,152 @@
+};
+
+// usereducer finctions
+const GymAccessoriesReduceFun = (state, action) => {
```
here filter and sort function is repeating so putting in one file and access it from there will reduce your work and your code look nice

Akshaay9 replied:
Hey Shivam
Thank you for your review,
I really wanted to do that, i have individual filter since i dont want to loose filtered state of one category if user apply filter on other category/pages since all filtered state will be depend on one function so i wanted to persist that  state and storing it in localstorage would make that heavy so,
Thank you
Akshay

shivamupadhyay008 reviewed (commented):
hey akshay its amazing work you did your code looks clean and your hard work reflects here so some minor change that i found that can improve more readability of your code there wise work is  great

Akshaay9 committed:
fixed UI bug of skeleton loader and updated few products images"

Akshaay9 committed:
Merge branch 'development' of https://github.com/Akshaay9/Ecommerce_Project into development

Akshaay9 committed:
Fixed minor UI bugs and made changes to filter product tags

Akshaay9 committed:
updated the background image of the product page

Akshaay9 committed:
FIxed the ui bug on mobile touch

Akshaay9 committed:
Made UI changes to Filter bar onHoover

uditkumar01 commented:
hi akshay, everything looks perfect. I have only one suggestion that change you favicon though its not that important.

Akshaay9 committed:
feat:added initial ping

Akshaay9 committed:
style:fixed height of products

Akshaay9 committed:
css:remooved unused CSS

Akshaay9 committed:
Updated readme.md

Akshaay9 committed:
Update image of readme.MD

Akshaay9 committed:
feat:added guest user login functionality

Akshaay9 committed:
Merge branch 'development' of https://github.com/Akshaay9/Ecommerce_Project into development
